sqlalchemy-stubs/engine/result.pyi

```diff
@@ -1,114 +1,134 @@
+_TMappingResult = TypeVar("_TMappingResult", bound=MappingResult)
+_TChunkedIteratorResult = TypeVar(
+    "_TChunkedIteratorResult", bound=ChunkedIteratorResult
+)
 
 class ResultMetaData:
     @property
-    def keys(self): ...
+    def keys(self) -> RMKeyView: ...
 
-class RMKeyView(collections_abc.KeysView):
+class RMKeyView(collections_abc.KeysView[str]):
     def __init__(self, parent: Any) -> None: ...
-    def __len__(self): ...
-    def __iter__(self) -> Any: ...
-    def __contains__(self, item: Any): ...
-    def __eq__(self, other: Any) -> Any: ...
-    def __ne__(self, other: Any) -> Any: ...
+    def __len__(self) -> int: ...
+    def __iter__(self) -> Iterator[str]: ...
+    def __contains__(self, item: object) -> bool: ...
+    def __eq__(self, other: Any) -> bool: ...
+    def __ne__(self, other: Any) -> bool: ...
 
 class SimpleResultMetaData(ResultMetaData):
     def __init__(
         self,
-        keys: Any,
+        keys: Iterable[str],
         extra: Optional[Any] = ...,
         _processors: Optional[Any] = ...,
         _tuplefilter: Optional[Any] = ...,
         _translated_indexes: Optional[Any] = ...,
         _unique_filters: Optional[Any] = ...,
     ) -> None: ...
 
-def result_tuple(fields: Any, extra: Optional[Any] = ...): ...
+def result_tuple(fields: Any, extra: Optional[Any] = ...) -> Row: ...
 
 class ResultInternal(InPlaceGenerative): ...
 
 class _WithKeys:
-    def keys(self): ...
+    def keys(self) -> RMKeyView: ...
 
 class Result(_WithKeys, ResultInternal):
     def __init__(self, cursor_metadata: Any) -> None: ...
-    def yield_per(self, num: Any) -> None: ...
-    def unique(self, strategy: Optional[object] = ...) -> Result: ...
-    def columns(self, *col_expressions: object) -> Result: ...
-    def scalars(self, index: Int = ...) -> ScalarResult: ...
-    def mappings(self): ...
-    def __iter__(self) -> Any: ...
-    def __next__(self): ...
-    def next(self): ...
-    def partitions(self, size: Optional[Int] = ...) -> Iterator[List[Row]]: ...
+    def yield_per(self: _TResult, num: Any) -> _TResult: ...
+    def unique(
+        self: _TResult, strategy: Optional[object] = ...
+    ) -> _TResult: ...
+    def columns(self: _TResult, *col_expressions: object) -> _TResult: ...
+    def scalars(self, index: int = ...) -> ScalarResult: ...
+    def mappings(self) -> MappingResult: ...
+    def __iter__(self) -> Iterator[Row]: ...
+    def __next__(self) -> Row: ...
+    if sys.version_info < (3, 0):
+        def next(self) -> Row: ...
+    def partitions(self, size: Optional[int] = ...) -> Iterator[List[Row]]: ...
     def fetchall(self) -> List[Row]: ...
-    def fetchone(self) -> Row: ...
-    def fetchmany(self, size: Optional[Int] = ...) -> List[Row]: ...
+    def fetchone(self) -> Optional[Row]: ...
+    def fetchmany(self, size: Optional[int] = ...) -> List[Row]: ...
     def all(self) -> List[Row]: ...
-    def first(self) -> Row: ...
```

Comment:
separately, we need to fix all the comment annotations in the main source for these

Reply:
should we just remove them?

Reply:
Yeah, the comment annotations should be removed since we're using stubs now

Reply:
added an issue https://github.com/sqlalchemy/sqlalchemy/issues/6207